models/case.js

```diff
@@ -3,9 +3,37 @@ module.exports = (sequelize, DataTypes) => {
     status: DataTypes.STRING,
     helpline: DataTypes.STRING,
     info: DataTypes.JSONB,
+    twilioWorkerId: DataTypes.STRING,
   });
 
-  Case.associate = models => Case.hasMany(models.Contact, { foreignKey: 'caseId' });
+  Case.associate = models => {
+    Case.hasMany(models.Contact, { foreignKey: 'caseId' });
+    Case.hasMany(models.CaseAudit, { foreignKey: 'caseId' });
+  };
+
+  Case.afterUpdate('auditCaseHook', async (caseInstance, options) => {
+    const contacts = await caseInstance.getContacts();
+    const contactsId = contacts.map(contact => contact.dataValues.id);
+    const previousValue = {
+      // eslint-disable-next-line no-underscore-dangle
+      ...caseInstance._previousDataValues,
+      contacts: contactsId,
```

Comment:
Right, an update on `Case` itself wouldn't change the associated contacts, since this information resides in the `Contact` table. You're also right when you say a contact change would be tracked by the hook in the `Contact` model.